engines/ic3ia.cpp

```diff
@@ -341,6 +340,45 @@ bool IC3IA::is_global_label(const Term & l) const
   return true;
 }
 
+void IC3IA::reabstract()
+{
+  // predicates from init and bad
+  UnorderedTermSet preds;
+  get_predicates(solver_, ts_.init(), preds, false, false, true);
```

Comment:
One thing to note here. It looks a little funny to me to get the predicates from init and bad before calling `ia_.do_abstraction()`. It doesn't actually matter because this abstraction doesn't change init or bad, but just to make it more intuitive we could consider moving that after the re-abstraction. I don't feel strongly either way though.

Reply:
updated in e4d900f